optimum/amd/ryzenai/pipelines/__init__.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+# Licensed under the MIT License.
+
+
+from typing import TYPE_CHECKING, Any, Optional, Union
+
+from optimum.exporters import TasksManager
+from transformers import ImageClassificationPipeline, Pipeline, PretrainedConfig
+from transformers import pipeline as transformers_pipeline
+from transformers.image_processing_utils import BaseImageProcessor
+from transformers.onnx.utils import get_preprocessor
+
+from ..modeling import RyzenAIModel, RyzenAIModelForImageClassification, RyzenAIModelForObjectDetection
+from ..models import YoloV3ImageProcessor, YoloV5ImageProcessor, YoloV8ImageProcessor, YoloXImageProcessor
+from .image_classification import TimmImageClassificationPipeline
+from .object_detection import YoloObjectDetectionPipeline
+
+
+if TYPE_CHECKING:
+    from transformers.feature_extraction_utils import PreTrainedFeatureExtractor
+
+pipeline_map = {
+    "yolox": {"preprocessor": YoloXImageProcessor, "impl": YoloObjectDetectionPipeline},
+    "yolov5": {"preprocessor": YoloV5ImageProcessor, "impl": YoloObjectDetectionPipeline},
+    "yolov3": {"preprocessor": YoloV3ImageProcessor, "impl": YoloObjectDetectionPipeline},
+    "yolov8": {"preprocessor": YoloV8ImageProcessor, "impl": YoloObjectDetectionPipeline},
+}
+
+RYZENAI_SUPPORTED_TASKS = {
+    "image-classification": {
+        "impl": ImageClassificationPipeline,
+        "class": (RyzenAIModelForImageClassification,),
+        "default": "amd/resnet50",
+        "type": "image",
+    },
```

Comment:
pipelines.mdx says there's no default?

Reply:
This default model `amd/resnet50` does not work for now without passing the correct `image-processor`. The user needs to pass the `image_processor` with `data_format NHWC` for most of Ryzen pretrained models. 
For object detection, I have created preprocessors so I have taken care of it.

Reply:
It appears you updated pipelines.mdx accordingly.

Reply:
I do throw an error when user tries to create a default model for the task. Possibly will soon update with a model to use for default